controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -99,6 +99,9 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, contro
 	}
 
 	if controlPlane.KCP.Status.ReadyReplicas > 0 {
+		// NOTE: this field is part of the Cluster API contract and it is used to orchestrate provisioning.
+		// The value of this field is never updated after provisioning is completed. Please use conditions
+		// to check the operational state of the control plane.
 		controlPlane.KCP.Status.Ready = true
```

Comment:
I think this doc should be added in the definition of field controlPlane.KCP.Status.Ready: https://github.com/kubernetes-sigs/cluster-api/blob/785ffa4722f2da7f35cb03436b0de6454b78f7bb/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go#L273-L276

Reply:
Thanks for pointing this out, I've updated in the above mentioned file.

Reply:
Thanks. This also applies to the comment for field Cluster.status.ControlPlaneReady

Reply:
I've updated the godocs for cluster_types with the latest commit.